### PR TITLE
*: replace interface with Datum in `ast.Row`.

### PR DESCRIPTION
**Pull request by coocood**

### CODE REVIEW

**Comment by coocood** [comment]
@shenli @zimulala PTAL

**Comment by shenli** [comment]
LGTM

File: driver.go [hunk 1]
```diff
@@ -452,47 +452,42 @@ func (r *driverRows) Next(dest []driver.Value) error {
...
-		case uint64:
-			dest[i] = uint64(v)
-		case uint:
-			dest[i] = uint64(v)
-		case mysql.Duration:
-			dest[i] = v.String()
-		case mysql.Time:
+		switch xi.Kind() {
+		case types.KindNull:
+			dest[i] = nil
```
Comment (zimulala) [inline driver.go]: can we use `xi.GetValue()`?

File: util/types/datum.go [hunk 2]
```diff
@@ -250,7 +266,7 @@ func (d *Datum) compareDatum(ad Datum) int {
 		return d.compareInt64(ad.GetInt64())
 	case KindUint64:
 		return d.compareUint64(ad.GetUint64())
-	case KindFloat64:
+	case KindFloat32, KindFloat64:
```
Comment (zimulala) [inline util/types/datum.go]: if it has KindFloat32, why doesn't have KindInt32

Comment (coocood) [inline util/types/datum.go]: Float32 produces different prints value than Float64, but Int32 prints same value as Int64.

**Comment by zimulala** [comment]
LGTM

**Pull request merged**